pghoard/rohmu/object_storage/azure.py

```diff
@@ -165,59 +182,61 @@ def _stream_blob(self, key, fileobj, progress_callback):
         allows reading entire blob into memory at once or returning data from random offsets"""
         file_size = None
         start_range = 0
-        chunk_size = self.conn.MAX_CHUNK_GET_SIZE
+        chunk_size = self.conn._config.max_chunk_get_size  # pylint: disable=protected-access
         end_range = chunk_size - 1
         while True:
             try:
                 # pylint: disable=protected-access
-                blob = self.conn._get_blob(self.container_name, key, start_range=start_range, end_range=end_range)
+                blob = self.conn.get_blob_client(self.container_name, key)
+                download_stream = blob.download_blob(offset=start_range, length=end_range)
```

Comment:
Is the length parameter named confusingly and it is in fact end offset? If not then the current implementation is incorrect and this should just be passing chunk_size instead of ever increasing size. If the name is confusing this needs to have a comment telling about that.